### PR TITLE
Fix CI testcase failure: TestMultiZkConnection

### LINKED ISSUE #2414: [Failed CI Test] afterClass(org.apache.helix.integration.multizk.TestMultiZkConnectionConfig)
This issue is created for tracking unstable test: afterClass(org.apache.helix.integration.multizk.TestMultiZkConnectionConfig)

### PR DESCRIPTION
**Pull request by desaikomal**
### Issues

- [X] My PR addresses the following Helix issues and references them in the PR description:
Fixes #2414 

### Description

- [x] Here are some details about my PR
We have seen afterClass(org.apache.helix.integration.multizk.TestMultiZkConnectionConfig) failing. The possible cause is that TestMultiZkHelixJavaApis re-initializes the _zkClient connection. Parent class TestMultiZkConnectionConfig already creates _zkClient in beforeClass().

So it is possible that we have dangling zkclient which in the afterClass cleanup causes the failure.

### Tests

- [ ] The following tests are written for this issue:

(List the names of added unit/integration tests)

- The following is the result of the "mvn test" command on the appropriate module:

(If CI test fails due to known issue, please specify the issue and test PR locally. Then copy & paste the result of "mvn test" to here.)

Testing Done:
[ERROR] Failures:
[ERROR]   TestNoThrottleDisabledPartitions.testDisablingTopStateReplicaByDisablingInstance:98 expected:<false> but was:<true>
[ERROR]   TestNoThrottleDisabledPartitions.testNoThrottleOnDisabledInstance:231->setupEnvironment:317->setupCluster:436 » ZkClient
[ERROR]   TestRecurringJobQueue.stopDeleteJobAndResumeRecurrentQueue:123 » Helix Failed ...
[ERROR]   TestRecurringJobQueue.testDeletingRecurrentQueueWithHistory:298 expected:<true> but was:<false>
[ERROR]   TestClusterStatusMonitorLifecycle.testClusterStatusMonitorLifecycle:297 Remaining MBeans:
...
[ERROR] Tests run: 1323, Failures: 5, Errors: 0, Skipped: 0
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:3.0.0-M3:test (default-test) on project helix-core: There are test failures.
[ERROR] 

### Changes that Break Backward Compatibility (Optional)

- My PR contains changes that break backward compatibility or previous assumptions for certain methods or API. They include:

(Consider including all behavior changes for public methods or API. Also include these changes in merge description so that other developers are aware of these changes. This allows them to make relevant code changes in feature branches accounting for the new method/API behavior.)

### Documentation (Optional)

- In case of new functionality, my PR adds documentation in the following wiki page:

(Link the GitHub wiki you added)

### Commits

- My commits all reference appropriate Apache Helix GitHub issues in their subject lines. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Code Quality

- My diff has been formatted using helix-style.xml 
(helix-style-intellij.xml if IntelliJ IDE is used)

### CODE REVIEW

**Comment by desaikomal** [comment]
No @junkaixue - it is not about log statement. this was existing code. i just removed the _zkClient overwritten by beforeClass()

**Review by junkaixue — APPROVED** [review]
Ah... Yeah, i missed the extension part. Good callout and thanks for fixing it!

File: helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkHelixJavaApis.java [hunk 1]
```diff
@@ -86,7 +86,7 @@ public class TestMultiZkHelixJavaApis extends TestMultiZkConnectionConfig {
 
   @BeforeClass
   public void beforeClass() throws Exception {
-    super.beforeClass();
+    super.beforeApiClass();
```
Comment (rahulrane50) [inline helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkHelixJavaApis.java:89]: If my understanding is correct then you are bypassing _zkClient object creation from TestMultiZkConnectionConfig class. If that's true then we should add few null checks in TestMultiZkConnectionConfig.setupCluster or other methods. This is just to make sure that any user of TestMultiZkConnectionConfig now expects to initialize _zkClient object.

Comment (rahulrane50) [reply helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkHelixJavaApis.java:89]: Another point, i think we are not even using _zkClient anywhere in the test here right? if not we can completely remove it from this test.

**Review by rahulrane50 — APPROVED** [review]
Thanks for fixing this Komal!
Just one thought, now there can be also paths about not initializing _zkClient anywhere in beforeClass paths. May be we can add null checks in afterClass of MultiZkTestBase to save us from few failures.

File: helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkHelixJavaApis.java (hunk 1 above)
Comment (desaikomal) [reply helix-core/src/test/java/org/apache/helix/integration/multizk/TestMultiZkHelixJavaApis.java:89]: we are using indirectly through base class. so i can't remove.

**Comment by desaikomal** [comment]
the same test cases are failing. so the fix is not correct. let me read the code again.

**Pull request closed without merging**